packages/client/src/internals/TRPCUntypedClient.ts

```diff
@@ -52,7 +54,28 @@ type CreateTRPCClientBaseOptions<TRouter extends AnyRouter> =
          * @link https://trpc.io/docs/data-transformers
          **/
         transformer?: ClientDataTransformerOptions;
-      };
+      }) &
+    (TRouter['_def']['_config']['namespaceDelimiter'] extends DefaultNamespaceDelimiter
+      ? {
+          /**
+           * The delimiter used to separate namespaces and procedure names in router paths
+           * of child routers.
+           *
+           * You must use the same namespaceDelimiter on the backend and frontend
+           * @link https://trpc.io/docs/merging-routers
+           */
+          namespaceDelimiter?: 'You must set a namespaceDelimiter on the backend router';
```

Comment:
```suggestion
          namespaceDelimiter?: TRouter['_def']['_config']['namespaceDelimiter'];
```

Reply:
> but failing CI

Oops, forgot to add the new prop to links tests.

---

Regarding the suggestion, good catch! 

Shouldn't it be just `DefaultNamespaceDelimiter` in this case?